pkg/api/pod/util.go

```diff
@@ -391,6 +391,20 @@ func dropDisabledFields(
 		})
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) && !lifecycleTypeInUse(oldPodSpec) {
```

Comment:
This is a bit different than other APIs, which explicitly nil-out the fields when gated.

Is there a reason not to do that here?  I get that it makes code a bit simpler to assume a value, but strictly speaking, the field more or less doesn't exist when gated.

In other words, is this PR adding a GA field with an alpha value or an alpha field?

Reply:
no I think what you said makes sense. The field itself is alpha not the value so setting type to nil seems sensible. Will fix